src/heartbeats.cpp

```diff
@@ -494,52 +574,65 @@ void MegaBackupMonitor::onSyncStateChanged(MegaApi* /*api*/, MegaSync *sync)
     updateOrRegisterSync(sync);
 }
 
-std::shared_ptr<HeartBeatSyncInfo> MegaBackupMonitor::getHeartBeatBackupInfoByTransfer(MegaTransfer *transfer)
+void MegaBackupMonitor::onSyncDeleted(MegaApi *api, MegaSync *sync)
 {
-    if (!transfer->isSyncTransfer())
+    auto hBPair = mHeartBeatedSyncs.find(sync->getTag());
+    if (hBPair != mHeartBeatedSyncs.end())
     {
-        return nullptr;
-    }
-
-    int syncTag = 0;
+        mClient->reqs.add(new CommandBackupRemove(mClient, hBPair->second->backupId()));
 
-    // use map to get the syncTag directly if there was one
-    auto mTSPair = mTransferToSyncMap.find(transfer->getTag());
-    if (mTSPair != mTransferToSyncMap.end())
-    {
-        syncTag = mTSPair->second;
+        mHeartBeatedSyncs.erase(hBPair); //This is speculative: could be moved to backupremove_result
+        // in case we wanted to handle possible failing cases.
```

Comment:
It would be an edge case, but... if we speculative remove it and the API doesn't process the removal, it will create an inconsistent state where the API returns a backup to the backup centre, but no client is heartbeating it.
I think it would be better to not do it speculative, not because of failures (which I think won't happen), but for a proper management of cases where the client is offline, in example, and gets closed before sending out the command.

Reply:
I see. The spec does mention that the apps should try to "do their best to inform the API if a sync is deleted/permanently removed". I wonder if that example of yours fits that description. There could be some rules in the API that can easily remove syncs after 30 mins of clients that are otherwise alive.. Anyway, having the sync removed from mHeartBeatedSyncs speculatively or only upon successful API return, won't fix the proposed use case. After `onSyncDeleted` callbacks are called, the SDK no longer knows about the sync. Fixing that will entail caching some `to_be_deleted_backups`, or caching some `unresolved_requests`. Doable, but would it be worth it?

Reply:
Definitely not worthy for an edge case like this one. Let's assume that the inconsistency might happen in an adverse scenario. In the end, the same thing can happen when the user uninstall the app without the change to unregister the sync, so the Backup Centre UI will need to consider those orphaned backups at some point and provide a way to get rid of them.
Thanks